### DISCUSSION TITLE
If it is possible to dump the compiled wasm code of a Python script?

### DISCUSSION THREAD

**Discussion by yayuwang0624** [pyodide/pyodide#3828] [Q&A] [answered]
If I understand correctly, pyodide first translates a python script to web assembly and integrated it into js. I wonder if it is possible to get the wasm representation of the python script (without WASI)?

Thanks

**Comment by hoodmane** [comment, marked as answer]
Pyodide executes Python code in the normal way that CPython does: the Python compiler to compiles the Python source code into Python bytecode and then the Python interpreter runs the bytecode. So no you can't use Pyodide to translate Python to wasm code. It is possible to get the Python bytecode, but that is probably less interesting to you.

**Reply by yayuwang0624** [reply]
got it, thank you
